packages/block-editor/src/components/link-control/index.js

```diff
@@ -235,6 +243,29 @@ function LinkControl( {
 		}
 	};
 
+	const resetInternalValues = () => {
+		setInternalUrlInputValue( value?.url );
+		setInternalTextInputValue( value?.title );
+	};
+
+	const handleCancel = ( event ) => {
+		event.preventDefault();
+		event.stopPropagation();
+
+		// Ensure that any unsubmitted input changes are reset.
+		resetInternalValues();
+
+		if ( hasLinkValue ) {
+			// If there is a link then exist editing mode and show preview.
+			stopEditing();
+		} else {
+			// If there is no link value, then remove the link entirely.
+			onRemove?.();
```

Comment:
I am now uncertain as to whether this is a good change. 

We are assuming that "cancel" when there is no link means "remove the link". But is that a decision the _component_ should be making?

Perhaps rather we should simply be calling `onCancel` and allow the _consumer_ to decide how that should be handled. I think it's _likely_ that 99% of the time the passed `onCancel` handler will basically be the same as `onRemove` but should we be _assuming_ that?

Reply:
It feels like a fair assumption to me.

Reply:
```suggestion
			onCancel?.();
```

We could do this but make the default value of `onCancel` be `onRemove`. That way we default to `onRemove` but still allow the consumer to optionally change that if they wish.

Reply:
Let's leave as is.